File: compareEnv.py
def compareEnv(a, b):
    mapA = {}
    mapB = {}

    for i in range(len(a)):
        temp = a[i].split("=")
        mapA[temp[0]] = temp[1]

    for i in range(len(b)):
        temp = b[i].split("=")
        mapB[temp[0]] = temp[1]

    aKeys = mapA.keys()
    bKeys = mapB.keys()
    res = []

    for i in range(len(aKeys)):
        if aKeys[i] not in mapB:
            res.append(aKeys[i] + ' is only in A')
        elif mapB[aKeys[i]] != mapA[aKeys[i]]:
            res.append(aKeys[i] + ' is different')

    for i in range(len(bKeys)):
        if bKeys[i] not in mapA:
            res.append(bKeys[i] + ' is only in B')
    res.sort()
    return res

File: compareEnv.js
function compareEnv(a, b) {
    let mapA = {};
    let mapB = {};

    a.map(e => {
        let temp = e.split('=');
        mapA[temp[0]] = temp[1]
    })

    b.map(e => {
        let temp = e.split('=');
        mapB[temp[0]] = temp[1]
    })

    let res = []
    let aKeys = Object.keys(mapA);
    let bKeys = Object.keys(mapB);

    aKeys.map(key => {
        if (!mapB[key]) {
            res.push(`${key} is only in A`);
        } else if (mapA[key] !== mapB[key]) res.push(`${key} is different`);
    })

    bKeys.map(key => {
        if (!mapA[key]) {
            res.push(`${key} is only in B`);
        }
    })

    return res.sort();
}